suggestions (#767)
neicnordic/sensitive-data-archive: sda-download/biohackathon/paper.md

umbrella called the Sensitive Data Archive to support these efforts.

This project focused on the data access part of the infrastructure. The
files are encrypted in the archives, using the crypt4gh standard [@crypt4gh].

> Currently, we have a data access processes, where the files are either

Reviewer: ```suggestion
Currently, we have a data access process, where the files are either
```

Reviewer 2: fixed

# Results

> In order to achieve the project goals, both htsget-rs and sda-download needed to be extened.

Reviewer: `extended` on the suggestion?

The following sections describe the work done on each of the services.

## HTSGet

In order to enable for random data access on encrypted files, we

> extended htsget-rs [@htsget-rs] to work with the new version of the sda-download.

Author: which new version? the one that is described below?

Reviewer 2: the version that it was developed during the biohackathon

We also extended the functionality of the sda-download service to support re-encryption of
requested files. This allows users to get files that are encrypted with their own
keypair instead of receiving just plain unencrypted files. To ensure the security of

> the archive secret key (avoid to keep it in a service that is available directly from the is available directly from the

Author: ```suggestion
the archive secret key (that is, to avoid keeping it in the htsget service, which is available directly from the
```
is this rephrased version correct?

internet) we have implemented a small microservice (gRPC Server in the diagram)

> that recieves the encrypted header and a public key, re-encrypts the file, and sends

Reviewer: ```suggestion
that receives the encrypted header and a public key, re-encrypts the file, and sends
```

Reviewer 2: fixed

 ```

## Additional updates

Two docker compose files were developed in order to include the whole infrastructure for the sensitive data archive and the htsget-rs version. These compose files allow the users to make requests to the htsget and receive a response containing pointers to the requested parts of the data, which the client can then download and assemble.

The Sensitive Data Archive Command Line Interface (sda-cli) has a demo version of the htsget command.

> This command, when provided with the public key of a user and (potentially) a region for a file, it makes the request to htsget, retrieves the results, requests the particular regions from the sda-download and then assembles these parts together. The end result should be a file that is encrypted with the user's public key, containing the regions requested by the user.

Reviewer 3: ```suggestion
This command, when provided with the public key of a user and (potentially) a region for a file, makes the request to htsget, retrieves the results, requests the particular regions from the sda-download and then assembles these parts together. The end result should be a file that is encrypted with the user's public key, containing the regions requested by the user.
```

# Conclusions and Future work

Some of the collaborators that joined our project were also members of the team developing the
htsget protocol for bioinformatics in Rust from the University of Melbourne (Center for Cancer Research).
This in person collaboration helped identifying the needs to achieve the project goal,
and to closely work together on developing the sda-download and
htsget server to meet those needs.

The project was focused on improving the data access part of the sensitive data archive infrastructure. The existing process
of data access was cumbersome: user receiving the whole file even when only a small part of it was needed.
To address this, the htsget-rs was extended to support access to encrypted data and it is planned to extend it further

> by implementing the crypt4gh and enabling support for crypt4gh edit lists. The sda-download service was also

Author: what does "implementing the crypt4gh" mean here..?

Reviewer 2: fixed

extended to support re-encryption of requested files and we further aim to
implement authentication of requests/tickets. This will allow the sda-download
service to fully trust that a request comes from the htsget-rs service, as well as
to return only the requested parts of the file based on the calculations done on the htsget-rs side. We also aim for a showcase where an
encrypted file will be read over the htsget protocol.

> Furthermore, there are two additional focus areas for future development. First, we plan to implement random access of encypted files in sda-download service.

Reviewer 3: ```suggestion
Furthermore, there are two additional focus areas for future development. First, we plan to implement random access of encrypted files in sda-download service.
```

Secondly, we aim to extend the sda-cli to automate

> the key genaration for communication with htsget, by automatically generate a

Reviewer 3: ```suggestion
the key generation for communication with htsget, by automatically generate a
```
